src/ReflectionMethod.php

```diff
@@ -52,7 +52,8 @@ public function __construct(
         ReflectionClass $declaringClass = null
     ) {
         //for some reason, ReflectionMethod->getNamespaceName in php always returns '', so we shouldn't use it too
-        $this->className        = $className;
+        $this->className        = is_object($className) ? get_class($className) : ltrim($className, '\\');
```

Comment:
Does this class even support passing an object in `$className` parameter? I'm suspecting it doesn't, because parameter name suggests that. Yeah, I know that PHP built-in reflection supports that.

If it doesn't, then do `ltrim` all the time.

Reply:
\Go\ParserReflection\ReflectionClass supports (documented like string|object) and is_object check present. I decide 'casting' object to string in \Go\ParserReflection\ReflectionMethod & \Go\ParserReflection\ReflectionProperty make them more consist with built-in & \Go\ParserReflection\ReflectionClass and provide expected behavior. Do you agree?

Reply:
I understand now. Since library only supports class names and not objects you've created workaround code to convert object (if somebody gives it by mistake) into a class so that library won't have any issues.

Nice catch, but this (and associated place in `ReflectionProperty` and other places) needs to be moved into separate PR with corresponding tests. I'd better not to mix this with topic of this PR.